test-helpers/src/connection/cassandra.rs

```diff
@@ -118,66 +198,13 @@ pub enum CassandraConnection {
 }
 
 impl CassandraConnection {
-    pub async fn new(contact_points: &str, port: u16, driver: CassandraDriver) -> Self {
-        CassandraConnection::new_inner(contact_points, port, driver, None, None).await
-    }
-
-    pub async fn new_tls(
-        contact_points: &str,
-        port: u16,
-        ca_cert_path: &str,
-        driver: CassandraDriver,
-    ) -> Self {
-        match driver {
-            #[cfg(feature = "cassandra-cpp-driver-tests")]
-            CassandraDriver::Datastax => {
-                let ca_cert = read_to_string(ca_cert_path).unwrap();
-                let mut ssl = Ssl::default();
-                Ssl::add_trusted_cert(&mut ssl, &ca_cert).unwrap();
-
-                CassandraConnection::new_inner(
-                    contact_points,
-                    port,
-                    driver,
-                    None,
-                    Some(Tls::Datastax(ssl)),
-                )
-                .await
-            }
-            // TODO actually implement TLS for cdrs-tokio
-            CassandraDriver::CdrsTokio => todo!(),
-            CassandraDriver::Scylla => {
-                let mut context = SslContext::builder(SslMethod::tls()).unwrap();
-                context.set_ca_file(ca_cert_path).unwrap();
-                let ssl_context = context.build();
-
-                CassandraConnection::new_inner(
-                    contact_points,
-                    port,
-                    driver,
-                    None,
-                    Some(Tls::Scylla(ssl_context)),
-                )
-                .await
-            }
-        }
-    }
-
-    pub async fn new_with_compression(
-        contact_points: &str,
-        port: u16,
-        driver: CassandraDriver,
-        compression: Compression,
-    ) -> Self {
-        CassandraConnection::new_inner(contact_points, port, driver, Some(compression), None).await
-    }
-
-    async fn new_inner(
+    pub async fn new(
         contact_points: &str,
         port: u16,
         driver: CassandraDriver,
         compression: Option<Compression>,
         tls: Option<Tls>,
+        protocol: ProtocolVersion,
```

Comment:
the protocol version is unused by scylla.
When compression is unused by a backend we panic, should we do the same for the protocol?